### ISSUE TITLE
WPAPass not callable from the LLVM pass manager

### ISSUE THREAD

**Issue by pzieris** [SVF-tools/SVF#289]
Hi Yulei,

I have been using SVF for a while and just recently pulled in the newest version from github. I am using SVF by loading a shared lib in `opt` and calling the `WPAPass` through LLVM's pass manager. After updating, I noticed that in commit https://github.com/SVF-tools/SVF/commit/e02422d70b81a8ac1907d11a3ff5e38f767d405d you removed the LLVM module pass functionality from `WPAPass`, which is why it can no longer be called by the pass manager.

Now, my question is whether it would be okay for you if I re-add everything so that `WPAPass` is callable through the pass manger again and then submit a pull request?

**Comment by yuleisui** [comment]
Yes, you are welcomed to do so and also DDAPass if you can. But it would be good that your modifications are for these two passes only.

**Comment by pzieris** [comment]
Hi Yulei,

I made the necessary changes to the `WPAPass` and noticed that the `DDAPass` was still callable from the LLVM pass manager, but the `Module` was not correctly transformed to a `SVFModule`. I fixed the issue.

Also, for you information, while going through the sources, I noticed that the following passes also register as a LLVM pass. Not sure if that's intended (because you mentioned only to modify `WPAPass` and `DDAPass`...):

- `BreakConstantGEPs`
- `MergeFunctionRets`
- `MTA`
